bluetooth/src/androidLibMain/kotlin/device/DeviceConnectionManager.kt

```diff
@@ -195,12 +204,16 @@ internal actual class DeviceConnectionManager(
                 gatt.await().setCharacteristicNotification(action.characteristic.wrapper, action.enable).also {
                     info(TAG, "setCharacteristicNotification result: $it")
                 }
-                // Enable/Disable remote notifications
-                val value = if (action.enable) BluetoothGattDescriptor.ENABLE_NOTIFICATION_VALUE
-                else BluetoothGattDescriptor.DISABLE_NOTIFICATION_VALUE
+
+                info { " * wrapper property: ${action.characteristic.wrapper.properties}" }
+                val value = when {
+                    action.enable and action.characteristic.wrapper.hasProperty(PROPERTY_INDICATE) -> BluetoothGattDescriptor.ENABLE_INDICATION_VALUE
+                    action.enable and action.characteristic.wrapper.hasProperty(PROPERTY_NOTIFY) -> BluetoothGattDescriptor.ENABLE_NOTIFICATION_VALUE
```

Comment:
This will always prefer indicate over notify (AFAIK characteristics can have both). Is this the iOS behaviour we observed?

Also I think in this case for readability `&&` can be used (no need to avoid short-circuiting this evaluation)

Reply:
Can be indicate and notify for the same characteristic?

Reply:
I thought `NOTIFY` and `INDICATE` never come together; at least, it doesn't make sense to me, but you are right, technically it is possible (`0x30`).  On iOS `NOTIFY` has priority. Changed the order

Reply:
Also here, if acition.enable == false and doesn't contains property you will attempt to write disable which makes no sense

Reply:
@avdyushin just pushed an update. Please check